File: atcoder.jp/PyPy3 (7.3.0)/abc096/abc096_c.py
H, W = map(int,input().split())
s = [['.']+list(input())+['.'] for _ in range(H)]
s = [['.'] * (W+2)] + s + [['.'] * (W+2)]

ans = True
for i in range(1,H+1):
	for j in range(1,W+1):
		if s[i][j] == '#':
			if '#' not in [s[i+1][j],s[i-1][j],s[i][j+1],s[i][j-1]]:
				ans = False

print('Yes' if ans else 'No')

File: atcoder.jp/Nim (1.0.6)/abc096/abc096_c.nim
import strformat, strutils, sequtils, math, algorithm, tables, sets, lists, intsets, critbits, future

var
    H, W: int
(H, W) = stdin.readLine.split.map(parseInt)
let
    grid = (0..<H).mapIt(stdin.readLine)

var flag: bool
for i in 0..<H:
    for j in 0..<W:
        flag = false
        if grid[i][j] == '#':
            if i > 0:
                if grid[i - 1][j] == '#':
                    flag = true
            if j > 0:
                if grid[i][j - 1] == '#':
                    flag = true
            if i < H - 1:
                if grid[i + 1][j] == '#':
                    flag = true
            if j < W - 1:
                if grid[i][j + 1] == '#':
                    flag = true
            if not flag:
                echo "No"
                quit 0


echo "Yes"